instrumentation/nginx/src/otel_ngx_module.cpp

```diff
@@ -65,6 +66,62 @@ nostd::string_view NgxHttpFlavor(ngx_http_request_t* req) {
   }
 }
 
+static void NgxNormalizeAndCopyString(u_char* dst, ngx_str_t str) {
+  for (ngx_uint_t i = 0; i < str.len; ++i) {
+    u_char ch = str.data[i];
+    if (ch >= 'A' && ch <= 'Z') {
+      ch += 0x20;
+    } else if (ch == '-') {
+      ch = '_';
+    }
+
+    dst[i] = ch;
+  }
+
+  dst[str.len] = '\0';
+}
+
+static void OtelCaptureHeaders(nostd::shared_ptr<opentelemetry::trace::Span> span, ngx_str_t keyPrefix, ngx_list_t *headers,
+                        ngx_regex_t *sensitiveHeaderNames, ngx_regex_t *sensitiveHeaderValues,
+                        std::vector<ngx_table_elt_t*> excludedHeaders = {}) {
+  for (ngx_list_part_t *part = &headers->part; part != nullptr; part = part->next) {
+    ngx_table_elt_t *header = (ngx_table_elt_t*) part->elts;
+    for (ngx_uint_t i = 0; i < part->nelts; ++i) {
+      if (std::find(excludedHeaders.begin(), excludedHeaders.end(), &header[i]) != excludedHeaders.end()) {
+        continue;
+      }
+
+      u_char key[keyPrefix.len + header[i].key.len + 1]; 
+      NgxNormalizeAndCopyString((u_char*)ngx_copy(key, keyPrefix.data, keyPrefix.len), header[i].key);
+
+      bool sensitiveHeader = false;
+#if (NGX_PCRE)
+      if (sensitiveHeaderNames) {
+        int ovector[3];
+        if (ngx_regex_exec(sensitiveHeaderNames, &header[i].key, ovector, 0) >= 0) {
+          sensitiveHeader = true;
+        }
+      }
+      if (sensitiveHeaderValues && !sensitiveHeader) {
+        int ovector[3];
+        if (ngx_regex_exec(sensitiveHeaderValues, &header[i].value, ovector, 0) >= 0) {
+          sensitiveHeader = true;
+        }
+      }
+#endif
+
+      nostd::string_view value;
+      if (sensitiveHeader) {
+        value = "[REDACTED]";
+      } else {
+        value = FromNgxString(header[i].value);
+      }
+
+      span->SetAttribute((const char*)key, nostd::span<const nostd::string_view>(&value, 1));
```

Comment:
minor: the `strlen` used in the construction of `nostd::string_view` can be avoided if `key` length is passed in here (and it would get rid of the need for a zero terminator).